netbout-web/src/main/java/com/netbout/rest/MarkdownTxtmark.java

```diff
@@ -88,11 +106,44 @@ private static String formatLinks(final String txt) {
         matcher.appendTail(result);
         return result.toString();
     }
+
+    /**
+     * Insert two spaces at the end of string that needs line break to
+     * force creation of HTML line break. Skip code blocks marked with markers
+     * elements.
+     * @param txt Text to parse
+     * @param markers Markers to be used in the iteration order
+     * @return Text with Markdown-formatted line breaks outside code blocks
+     */
+    private static String makeLineBreakMarkedCode(final String txt,
+        final Iterator<String> markers) {
+        final String result;
+        if (markers.hasNext()) {
+            final String marker = markers.next();
+            final StringBuilder builder = new StringBuilder();
+            boolean code = false;
+            for (final String fragment : txt.split(marker, -1)) {
+                if (code) {
+                    builder.append(marker)
+                    .append(fragment)
+                        .append(marker);
+                } else {
+                    builder.append(makeLineBreakMarkedCode(fragment, markers));
```

Comment:
@krzyk The idea is to exclude code blocks separated with 3 chars sequence from adding extra spaces (with existing method `makeLineBreak()`). But the problem still exist also for blocks separated with 2 chars and one char. So decided to exclude them too. Either writing the same code (with copy/paste) in the `else` clause or another call to this method with second separator. A much elegant code would be without recursion but with functional interfaces we have in java 8, but of course, this is not under discussion.